### DISCUSSION TITLE
Multiple Passkeys per Platform User Account

### DISCUSSION THREAD

**Discussion by renchris** [MasterKale/SimpleWebAuthn#578] [General]
I am working and looking for guidance towards implementing multiple passkeys per User Account. 

User A signs up for their platform account for the initial time registering a passkey (ex. Touch ID) attached to the Device A (ex. laptop) that they are on.
User A wishes to register a secondary passkey (ex. Face ID) attached to another Device B (ex. phone) that they have.

I presume the high-level platform user flow and logic is:

- User A, from their signed-in platform account on Device A, prompts on the page UI that they wish to add a new passkey.
- User A is given either an email or URL magic link, a QR code, or an OTP, with a certain valid time limit, to open with on their second Device B that they wish to register their second passkey on.
- User A then opens the 'Add New Passkey on this Device' page on Device B from the corresponding magic link or code.
- The page on Device B then prompts User A for their new Device B passkey.
- The platform user account now has both Device A passkey (ie. laptop touch ID) and Device B passkey (ie. phone Face ID) as valid passkey sign-on credentials.

How do we do this with SimpleWebAuthN?

**Comment by MasterKale** [comment]
Hello @renchris, is an assumption here that no passkeys are being synced between devices? "Touch ID" and "Face ID" are mentioned, but honestly it's Apple devices that are most likely to be able to sync between devices, as opposed to e.g. a user with a Windows laptop and an Android phone. Whether syncing is in play I think will influence guidance on how to help get a user protected by passkeys across devices.

**Reply by renchris** [reply]
Hi Matthew, yes, correct, the assumption is that no passkey are being synced between devices.

**Reply by MasterKale** [reply]
Then I'd say the short version of any suggested flows for bootstrapping new device registration range from "let the user perform hybrid auth from the new device if they can", to "send a magic link/one-time-code/etc... they can use on the new device". The former is preferable because, as a WebAuthn-backed process it's phishing resistant. The use of a one-time passcode of any kind is more reliable to pull off but is also phishable; I'd suggest factoring that into your threat modeling as you figure out how best to proceed.

**Reply by renchris** [reply]
Okay. If the preferred method is to "let the user perform hybrid auth from the new device", how do we know that the new passkey registration on the new Device B, which hasn't had its initial sign-in yet, corresponds to the same platform user account that was created on Device A?

**Reply by MasterKale** [reply]
At a high level, something like this:

1. User attempts to sign in on **Device B**
2. User can successfully perform hybrid auth using **Device A** to scan the QR code displayed on Device B
4. Start an authenticated session for the user on Device B
5. Look at the credential ID and/or `userHandle` (and if [`authenticatorAttachment` in the auth response](https://w3c.github.io/webauthn/#dom-publickeycredential-authenticatorattachment) exists and is `"cross-platform"`) from **Step 2** then immediately prompt the user to register Device B's local platform authenticator (because it's better UX than having them to do hybrid all the time)
